File: WP1_data_generator_python/domain_change.py
import numpy as np

def domain_change(x,to_time):
    shape = x.shape
    y = np.zeros(shape, dtype=complex)
    if to_time:
        for i in range(shape[1]):
            Htemp = x[:,i]
            y[:,i] = np.fft.ifft(Htemp)
    else:
        for i in range(shape[1]):
            Htemp = x[:,i]
            y[:,i] = np.fft.fft(Htemp)
    return y

File: WP1_data_generator_matlab/domain_change.m
%clear all;

%load('30realH2048IDL2.mat')
function y = domain_change(x, to_time)
    shape = size(x);

    y = zeros(shape);
    if to_time 
        for i= 1:shape(2)
            Htemp = x(:,i);
            y(:,i) = ifft(Htemp);
        end
    else
        for i= 1:shape(2)
            Htemp = x(:,i);
            y(:,i) = fft(Htemp);
        end
    end
    
end